### ISSUE TITLE
Enhancement: Smart renaming of files

### ISSUE THREAD

**Issue by BillRaymond** [estruyf/vscode-front-matter#545]
**Is your feature request related to a problem? Please describe.**
Yes. See two common issues with posting

## Dates in filenames
When you create a new piece of content, FM requests the title, so you could type something like `my new post` and then FM creates a file for you, naming it `yyyy-MM-dd-my-new-post.md`. But I frequently put up future posts that should not be seen for days or even weeks in the future. While I can change the date in the YAML front matter, the filename does not change, making it hard to maintain the files.

## Titles in a post
When you create a new piece of content, you type a title and FM creates the file for you. However:
1. After you create the file FM provides a UI telling you if the title may be too long
2. After you type the title, you may decide to change it completely. For example, I did this just recently:
Original title: Digital transformation and the dance between agilty and structure in Government
Udpated title: Agile Development for Digital Government Services: Challenges and Success Factors
Updated title (once again): Digital transformation and agility in government
3. You then have to copy the new title, go to Explorer, locate the file and rename it manually

## Problem statement
* Your filename's date prefix may not match the isPublishDate for the post
* Your filename's title may not match the title in the post
* Users will have difficulty managing their content since the filename ultimately represents the date and title of the content. For example, sorting content will be more difficult

## Value statement
* Content creators often write future posts, so this will support a common use case scenario
* Content creators often change the dates and titles of their posts, so that will support a common use case scenario

**Describe the solution you'd like**
Ideally, there is a "Smart rename" option in FM that allows me to rename the file with the updated `datetime` that is designated as `isPublishDate` and also modify the title of the filename if that changed (I do not believe there is a `isTitle` boolean, but I assume FM always expects a title?).

Optionally, offer a UI to modify the file name manually, but follow all the business rules you use to create a filename. For example, if I rename a file and type:

`2023 10 03 Welcome to October!`

The filename will be:
`2023-10-03-welcome-to-october.md`

**Describe alternatives you've considered**

**Additional context**
The following image represents a common use case, where I create a post where FM assumes I am going to post on the creation date. FM assumes the filename remains the same despite changing the date and title in YAML.

<img width="925" alt="file-naming-issue" src="https://user-images.githubusercontent.com/15895828/226497221-6a0af2a0-5c1e-4368-8959-a26c41796049.png">

**Comment by estruyf** [comment]
It might be related to: https://github.com/estruyf/vscode-front-matter/issues/519

There is an old `frontMatter.taxonomy.alignFilename` setting, which needs to be moved to the content type level, as you have more flexibility over the file alignment.

By the way: Do you use the filename as the slug?

**Comment by BillRaymond** [comment]
No, as a matter of fact I do not use slugs at all. I know, I know. I just haven't gotten around to it.

**Comment by AdriandeLima** [comment]
Has there been any update on this issue? I also prep content in advance, so end up with my web pages having a date mismatch with the publish date.
Is there any reason I should keep the date in the web page title, this might be an easy solution, disable the date in the content title?

**Comment by estruyf** [comment]
In the latest beta, there is a new `smart rename` action added.

<img width="564" height="133" alt="Image" src="https://github.com/user-attachments/assets/172a1782-aacd-427b-a47d-3f754f74a374" />

It combines the publishing date + title for the filename. If you don't want the date in the filename, you'll have to change the global `frontMatter.templates.prefix` setting or define it on Content Type level: https://frontmatter.codes/docs/content-creation/placeholders#using-placeholders
